src/TestSuite/Fixture/FixtureManager.php

```diff
@@ -227,14 +227,15 @@ protected function _loadFixtures($test)
     protected function _setupTable($fixture, $db, array $sources, $drop = true)
     {
         $configName = $db->configName();
-        if ($this->isFixtureSetup($configName, $fixture)) {
+        $isFixtureSetup = $this->isFixtureSetup($configName, $fixture);
+        if ($isFixtureSetup) {
             return;
         }
 
         $table = $fixture->sourceName();
         $exists = in_array($table, $sources);
 
-        if ($drop && $exists) {
+        if (($drop && $exists) || ($exists && !$isFixtureSetup)) {
```

Comment:
When we hit that ``if`` check, ``isFixtureSetup`` is ``false`` everytime so if we add ``$drop`` to the check, we would end up with the two parts of the ``if`` statement being exactly the same (``$exists && $drop``). (Don't know if I'm being super clear here).

To address your concern, I'd rather we add a new part to check if ``$fixture->schema()`` returns an instance of ``Table`` to see if we can rebuild the table.

What do you think ?